server/apps/immich/src/api-v1/job/job.service.ts

```diff
@@ -0,0 +1,143 @@
+import {
+  exifExtractionProcessorName,
+  generateJPEGThumbnailProcessorName,
+  IMetadataExtractionJob,
+  IThumbnailGenerationJob,
+  IVideoTranscodeJob,
+  metadataExtractionQueueName,
+  QueueNameEnum,
+  thumbnailGeneratorQueueName,
+  videoConversionQueueName,
+  videoMetadataExtractionProcessorName,
+} from '@app/job';
+import { InjectQueue } from '@nestjs/bull';
+import { Queue } from 'bull';
+import { BadRequestException, Inject, Injectable } from '@nestjs/common';
+import { AllJobStatusResponseDto } from './response-dto/all-job-status-response.dto';
+import { randomUUID } from 'crypto';
+import { ASSET_REPOSITORY, IAssetRepository } from '../asset/asset-repository';
+import { AssetType } from '@app/database/entities/asset.entity';
+import { GetJobDto } from './dto/get-job.dto';
+import { JobStatusResponseDto } from './response-dto/job-status-response.dto';
+
+@Injectable()
+export class JobService {
+  constructor(
+    @InjectQueue(thumbnailGeneratorQueueName)
+    private thumbnailGeneratorQueue: Queue<IThumbnailGenerationJob>,
+
+    @InjectQueue(metadataExtractionQueueName)
+    private metadataExtractionQueue: Queue<IMetadataExtractionJob>,
+
+    @InjectQueue(videoConversionQueueName)
+    private videoConversionQueue: Queue<IVideoTranscodeJob>,
+
+    @Inject(ASSET_REPOSITORY)
+    private _assetRepository: IAssetRepository,
+  ) {
+    this.thumbnailGeneratorQueue.empty();
+    this.metadataExtractionQueue.empty();
+    this.videoConversionQueue.empty();
+  }
+
+  async startJob(jobDto: GetJobDto): Promise<number> {
+    if (jobDto.jobId === QueueNameEnum.THUMBNAIL_GENERATION) {
```

Comment:
For readability in this function, it might help to do a switch dase on `jobId` and extract the code for specific queues into separate functions.

Reply:
Done. Thank you for the suggestion